logstash-core/src/test/java/org/logstash/config/ir/compiler/OutputDelegatorTest.java

```diff
@@ -0,0 +1,247 @@
+package org.logstash.config.ir.compiler;
+
+import org.assertj.core.data.Percentage;
+import org.jruby.RubyArray;
+import org.jruby.RubyClass;
+import org.jruby.RubyFixnum;
+import org.jruby.RubyHash;
+import org.jruby.RubyString;
+import org.jruby.RubySymbol;
+import org.jruby.java.proxies.ConcreteJavaProxy;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.logstash.Event;
+import org.logstash.config.ir.RubyEnvTestCase;
+import org.logstash.execution.ExecutionContextExt;
+import org.logstash.instrument.metrics.NamespacedMetricExt;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.logstash.RubyUtil.EXECUTION_CONTEXT_CLASS;
+import static org.logstash.RubyUtil.NAMESPACED_METRIC_CLASS;
+import static org.logstash.RubyUtil.OUTPUT_DELEGATOR_CLASS;
+import static org.logstash.RubyUtil.RUBY;
+
+public class OutputDelegatorTest extends RubyEnvTestCase {
```

Comment:
@danhermann since this uses the Ruby environment, you have to exclude it from the `javaTests` goal in Gradle and add it to the `rubyTests` one explicitly (since it requires ruby files and a bundler install run).